common/lib/xmodule/xmodule/fields.py

```diff
@@ -252,3 +253,19 @@ def enforce_type(self, value):
             return value
 
         return self.from_json(value)
+
+
+class ScoreField(JSONField):
+    def from_json(self, value):
+        if value is None:
+            return value
+        if isinstance(value, Score):
+            return value
+
+        keys = value.keys()
+        if 'raw_earned' not in keys or 'raw_possible' not in keys:
```

Comment:
FWIW, `if 'raw_earned' not in value or 'raw_possible' not in value:` will do the same thing for a dict.

Also, if you're storing a great many of these things, you could also model it as a slightly smaller list `(e.g. [0,2])`, to avoid redundant keys.